config/api/src/main/java/org/jboss/windup/config/tags/Tag.java

```diff
@@ -0,0 +1,126 @@
+package org.jboss.windup.config.tags;
+
+
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+import org.apache.commons.collections4.set.UnmodifiableSet;
+
+/**
+ * Represents a tag. Determined by it's lowercased name.
+ * @author Ondrej Zizka
+ */
+public final class Tag
+{
+    private final String name;
+    /**
+     * Keeps the "contains" relation.
+     */
+    private final Set<Tag> containedTags = new HashSet<>();
```

Comment:
That's the thing - they are not parent-children - it might mislead people into thinking it's a tree, which it is not.